augur/filter.py

```diff
@@ -98,25 +98,29 @@ def filter_by_exclude_all(metadata):
     return set()
 
 
-def filter_by_exclude(metadata, excluded_strains):
+def filter_by_exclude(metadata, exclude_files):
     """Exclude the given set of strains from the given metadata.
 
     Parameters
     ----------
     metadata : pandas.DataFrame
         Metadata indexed by strain name
-    excluded_strains : set[str]
-        Set of strain names to exclude from the given metadata
+    exclude_files : list[str]
+        List of filenames with strain names to exclude from the given metadata
```

Comment:
I wonder whether we should exclude strains by individual files instead of multiple files. For example, the ncov workflow uses multiple exclude files and when we view the resulting log from augur filter, we don't know which file excluded each strain:

![BCECC81B-5D08-41C3-9C83-B1828472FB6E](https://user-images.githubusercontent.com/85372/126243061-9a10e043-d4ef-451a-870f-f042a6f745e7.png)

This information is helpful when we are trying to understand whether a strain was excluded by the autogenerated diagnostics' "to-exclude" file or the hardcoded exclude file.

Reply:
Are you asking if we should pare down the CLI to accept only a single file, or rework the internals of filter to be able to report the specific file responsible? The former seems like a step in the wrong direction to me; the latter seems eminently reasonable and possible (though I don't know the complicating factors yet).

Reply:
>  rework the internals of filter to be able to report the specific file responsible

This one. I'll put it on the checklist for [the latest PR](https://github.com/nextstrain/augur/pull/750).